app/models/rbac.rb

```diff
@@ -248,7 +248,33 @@ def self.find_targets_with_user_group_rbac(klass, scope, rbac_filters, find_opti
     end
   end
 
+  def self.group(user_or_group)
+    case user_or_group
+    when User
+      user_or_group.current_group
+    when MiqGroup
+      user_or_group
+    when NilClass
```

Comment:
Please keep the logic out of the models.
they do know ownership, but each object will have different logic.

First pass we're hard codeing parents / children
but in the future we'll change the tenant's options or the group's options on how to do that search. This all belongs in the "entitlement" record and not in the actual objects themselves